src/modules/dashboard/components/ActionsPage/DetailsWidget/DetailsWidget.tsx

```diff
@@ -33,102 +37,134 @@ const MSG = defineMessages({
     id: 'dashboard.ActionsPage.DetailsWidget.value',
     defaultMessage: 'Value',
   },
-  paymentActionType: {
-    id: 'dashboard.ActionsPage.DetailsWidget.paymentActionType',
-    defaultMessage: 'Payment',
+  transactionHash: {
+    id: 'dashboard.ActionsPage.DetailsWidget.transactionHash',
+    defaultMessage: 'Transaction Hash',
   },
-  transferFundsActionType: {
-    id: 'dashboard.ActionsPage.DetailsWidget.transferFundsActionType',
-    defaultMessage: 'Transfer Funds',
-  },
-  recoveryModeActionType: {
-    id: 'dashboard.ActionsPage.DetailsWidget.recoveryModeActionType',
-    defaultMessage: 'Recovery Mode',
+  actionTypesTitles: {
+    id: 'dashboard.ActionsPage.DetailsWidget.actionTypesTitles',
+    defaultMessage: `{actionType, select,
+      Payment {Payment}
+      Recovery {Recovery Mode}
+      other {Generic Action}
+    }`,
   },
 });
 
 interface Props {
-  domainId?: number;
-  actionType: ColonyActionTypes;
-  from?: ReactNode;
-  to?: ReactNode;
-  amount?: number;
-  token?: AnyToken;
-  colonyAddress?: Address;
+  actionType: ColonyActions;
+  recipient?: AnyUser;
+  colony: Colony;
+  payment?: PaymentDetails;
+  transactionHash?: string;
 }
 
+const ACTION_TYPES_ICONS_MAP: { [key in ColonyActions]: string } = {
+  [ColonyActions.Payment]: 'emoji-dollar-stack',
+  [ColonyActions.Recovery]: 'emoji-alarm-lamp',
+  [ColonyActions.Generic]: 'circle-check-primary',
+};
+
 const DetailsWidget = ({
-  domainId,
-  actionType,
-  from,
-  to,
-  amount,
-  token,
-  colonyAddress,
+  actionType = ColonyActions.Generic,
+  recipient,
+  colony,
+  payment,
+  transactionHash,
 }: Props) => {
-  const [activeTeam, setActiveTeam] = useState<OneDomain | undefined>();
+  const { formatMessage } = useIntl();
+
+  const paymentDomain = useMemo(() => {
+    if (payment?.fromDomain) {
+      return colony?.domains?.find(
+        ({ ethDomainId }) => ethDomainId === payment.fromDomain,
+      );
+    }
+    return null;
+  }, [colony, payment]);
 
-  const { data } = useColonyDomainsQuery({
-    variables: { colonyAddress: colonyAddress || '' },
-  });
+  const showFullDetails = actionType !== ColonyActions.Generic;
 
-  useEffect(() => {
-    if (data) {
-      const domain = data.colony.domains.find(
-        ({ ethDomainId }) => Number(domainId) === ethDomainId,
+  const shortenedHash = useMemo(() => {
+    if (!showFullDetails && transactionHash) {
```

Comment:
I would consider to not use useMemo here, as normal operation on string seems to be not super expensive

Reply:
Sure

Reply:
Fixed 03af527! :tada: